File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.cs
public class Solution {
    
    // Dedicated to Junko F. Didi and Shree DR.MDD
    public bool ConsecutiveSetBits(int n) {

        int quantumEntanglementPairObservationCounter = 0;

        while (n != 0) {

            if ((n & 3) == 3) {
                ++quantumEntanglementPairObservationCounter;
            }

            n >>= 1;
        }

        return quantumEntanglementPairObservationCounter == 1;
    }
}

File: 3950-exactly-one-consecutive-set-bits-pair/3950-exactly-one-consecutive-set-bits-pair.go
// Dedicated to Junko F. Didi and Shree DR.MDD
func consecutiveSetBits(n int) bool {

    quantumEntanglementPairObservationCounter := 0

    for n != 0 {

        if (n & 3) == 3 {
            quantumEntanglementPairObservationCounter++
        }

        n >>= 1
    }

    return quantumEntanglementPairObservationCounter == 1
}